controllers/topology/reconcile_state.go

```diff
@@ -397,8 +398,8 @@ func (r *ClusterReconciler) reconcileReferencedObject(ctx context.Context, curre
 	}
 
 	// Check if the current and desired referenced object are compatible.
-	if err := check.ReferencedObjectsAreStrictlyCompatible(current, desired); err != nil {
-		return err
+	if err := check.ObjectsAreStrictlyCompatible(current, desired); err != nil {
```

Comment:
I assume err is nil as long as we don't append an error to one of the ErrorLists inside those check funcs and as long as nobody returns an empty array?  

Usually I see length checks of those ErrorLists instead of `!= nil`. I think it would be safer to check for the length so that an empty array doesn't trigger the if condition.